KsApi/models/GraphUserCreditCardTests.swift

```diff
@@ -0,0 +1,107 @@
+import XCTest
+@testable import KsApi
+
+final class GraphUserCreditCardTests: XCTestCase {
+
+  func testCreditCardsDecoding_noCards() {
+    let jsonString = """
+      {
+        "storedCards": {
+            "nodes": [],
+            "totalCount": 0
+        }
+      }
+      """
+    let data = jsonString.data(using: .utf8)
+
+    do {
+      //swiftlint:disable force_unwrapping
+      let cards = try JSONDecoder().decode(GraphUserCreditCard.self, from: data!)
+
+      XCTAssertEqual(cards.storedCards.nodes.count, 0)
+    } catch {
+      XCTFail("Failed to decode GraphUserCreditCard")
+    }
+  }
+
+  func testCreditCardsDecoding_hasCards() {
+    let jsonString = """
+        {
+          "storedCards": {
+            "nodes": [
+               {
+                  "expirationDate": "2023-02-01",
+                  "lastFour": "4242",
+                  "id": "3021",
+                  "type": "VISA"
+                },
+                {
+                  "expirationDate": "2020-02-01",
+                  "lastFour": "1111",
+                  "id": "2768",
+                  "type": "VISA"
+                }
+            ],
+            "totalCount": 0
+            }
+        }
+        """
+
+    let data = jsonString.data(using: .utf8)
+
+    do {
+      //swiftlint:disable force_unwrapping
+      let cards = try JSONDecoder().decode(GraphUserCreditCard.self, from: data!)
+
+      XCTAssertEqual(cards.storedCards.nodes.count, 2)
+
+      guard let firstCard = cards.storedCards.nodes.first else {
+        XCTFail("Failed to decode GraphUserCreditCard")
+        return
+      }
+
+      XCTAssertEqual(firstCard.type, GraphUserCreditCard.CreditCardType.visa)
+      XCTAssertEqual(firstCard.lastFour, "4242")
+      XCTAssertEqual(firstCard.expirationDate, "2023-02-01")
+      XCTAssertEqual(firstCard.id, "3021")
+    } catch {
+      XCTFail("Failed to decode GraphUserCreditCard")
+    }
+  }
+
+  func testCreditCardsDecoding_unknownCard() {
+    let jsonString = """
+        {
+          "storedCards": {
+            "nodes": [
+                {
+                  "expirationDate": "2020-02-01",
+                  "lastFour": "1111",
+                  "id": "2768",
+                }
+            ],
+            "totalCount": 0
+          }
+        }
+        """
+
+    let data = jsonString.data(using: .utf8)
```

Comment:
In future we should try to replace these with the non-failable initializer:
```swift
let data = Data(bytes: jsonString.utf8)
```
Then we won't need the force unwraps.